src/xbps/troubleshooting/common-errors.md

```diff
@@ -0,0 +1,45 @@
+# Common errors
+
+## Errors while updating or installing packages
+
+If there are any errors while updating or installing a new package, make sure
+that you are using the latest version of the remote repository index. Running
+[xbps-install(1)](https://man.voidlinux.org/xbps-install.1) with the `-S` option
+will guarantee that.
+
+### shlib errors
+
+If you get an error with the format:
+
+```
+libllvm8-8.0.1_2: broken, unresolvable shlib `libffi.so.6'
+```
+
+it is likely that orphan packages which have already been removed from the Void
+repos are still installed in your system. This can be solved by running
+[xbps-remove(1)](https://man.voidlinux.org/xbps-remove.1) with the `-o` option,
+which removes orphan packages.
+
+If you get an error message saying
+
+```
+Transaction aborted due to unresolved shlibs
+```
+
+the repositories are in the staging state, which can happen due to large builds.
+The solution is to wait for the builds to finish. You can view the builds'
+progress in the [Buildbot's Waterfall
+Display](https://build.voidlinux.org/waterfall).
+
+### repodata errors
```

Comment:
Uh, this paragraph gives impression that we break updates every quarter. We don't. Could you change to something like
> In March 2020 compression format of packages was changed from xz to zstd. If xbps wasn't updated to 0.54 (released June 2019) or newer, it's not possible to update system with it. [for searchability: ] Error message in that case is ... It is necessary to follow the steps ..."

, and remove or similarly rewrite and shorten paragraph in static.md?

Reply:
That's definitely not what I meant to imply. Thanks!

Reply:
@Chocimier ok, now I'm not too sure. repodata errors would be due to the repository information compression format having changed, not being able to unpack the packages themsevlves is a different issue. Do you know when the repository data format was changed? I can look for it a bit.

Is there a case XBPS will be able to understand the repo data information but not be able to unpack the packges?

Reply:
The problem with the error message for this issue is that there isn't any. I'm trying it out on a ROOTFS image from 2018, and it doesn't output anything unless you run the commands with the `-d` flag to get some debug output. With debug on, it does give some good output. I can try to include that. Anyway, the issue is with the repodata format more than the package themselves, because the failure happens earlier.

Reply:
Ah, right, it's repodata compression what causes problems here, not packages. Changes of the two was done same time, so it's _unlikely_ that xbps can handle one but not another.